*** /home/mks/mod/1.4.2/heaters.py	2025-07-05 08:07:12.272355325 +0000
--- heaters.py	2025-07-05 08:02:22.964187754 +0000
***************
*** 80,90 ****
              time_diff = read_time - self.last_temp_time
              self.last_temp = temp
              self.last_temp_time = read_time
-             self.control.temperature_update(read_time, temp, self.target_temp)
              temp_diff = temp - self.smoothed_temp
              adj_time = min(time_diff * self.inv_smooth_time, 1.)
              self.smoothed_temp += temp_diff * adj_time
              self.can_extrude = (self.smoothed_temp >= self.min_extrude_temp)
          #logging.debug("temp: %.3f %f = %f", read_time, temp)
      # External commands
      def get_pwm_delay(self):
--- 80,93 ----
              time_diff = read_time - self.last_temp_time
              self.last_temp = temp
              self.last_temp_time = read_time
              temp_diff = temp - self.smoothed_temp
              adj_time = min(time_diff * self.inv_smooth_time, 1.)
              self.smoothed_temp += temp_diff * adj_time
              self.can_extrude = (self.smoothed_temp >= self.min_extrude_temp)
+             # Use the smoothed temperature if smooth time is >1.0s, otherwise use
+             # the instantaneous temperature instead - stew675 - 2025/07/03
+             update_temp = (temp, self.smoothed_temp)[bool(self.smooth_time > 1.0)]
+             self.control.temperature_update(read_time, update_temp, self.target_temp)
          #logging.debug("temp: %.3f %f = %f", read_time, temp)
      # External commands
      def get_pwm_delay(self):
